test/functional/normal/jump_spec.lua

```diff
@@ -0,0 +1,47 @@
+local helpers = require('test.functional.helpers')(after_each)
+
+local clear = helpers.clear
+local command = helpers.command
+local eq = helpers.eq
+local funcs = helpers.funcs
+local feed = helpers.feed
+local write_file = helpers.write_file
+
+describe("jumplist", function()
+  before_each(clear)
+
+  it("doesn't add a new entry on startup", function()
+    eq("\n jump line  col file/text\n>", funcs.execute("jumps"))
+  end)
+
+  it("doesn't require two <C-O> strokes to jump back", function()
+    local fname1 = "Xtest-functional-normal-jump"
+    local fname2 = fname1 .. "2"
+    write_file(fname1, "first file contents")
+    write_file(fname2, "second file contents")
+
+    command("args " .. fname1 .. " " .. fname2)
+    local buf1 = funcs.bufnr(fname1)
+    local buf2 = funcs.bufnr(fname2)
+
+    command("next")
+    feed("<C-O>")
+    eq(buf1, funcs.bufnr("%"))
+
+    command("first")
+    command("snext")
+    feed("<C-O>")
+    eq(buf1, funcs.bufnr("%"))
+    feed("<C-I>")
+    eq(buf2, funcs.bufnr("%"))
+    feed("<C-O>")
+    eq(buf1, funcs.bufnr("%"))
+
+    command("drop " .. fname2)
+    feed("<C-O>")
+    eq(buf1, funcs.bufnr("%"))
+
+    os.remove(fname1)
+    os.remove(fname2)
```

Comment:
This would go in `after_each()` to avoid stale files if the test has an error that prevents it from completing.

In contrast, `write_file` stuff can stay at the start of the test, it does not need to go in `before_each()`.

Reply:
I'll fix it in the merge.